app/frontend/helpers/validators.ts

```diff
@@ -8,7 +8,8 @@ import {InternalErrorReason} from '../errors'
 
 const {ADALITE_MIN_DONATION_VALUE} = ADALITE_CONFIG
 const parseToLovelace = (str: string): Lovelace =>
-  Math.trunc(toCoins(parseFloat(str) as Ada)) as Lovelace
+  // Math.round to solve edge cases in floating number precision
+  Math.round(toCoins(parseFloat(str) as Ada)) as Lovelace
```

Comment:
I'd consider adding a unit test covering this helper function with the exact edge case(s) where the old implementation failed - it would serve as documentation and prevention of reintroduction of the same error in the future

Reply:
I added a test, tbh I find the comment more than self-explanatory, we can't explain the reason behind the errors in precision, nor we cant provide an example that explains it since they are basically infinite of them. I think anyone who would try to understand would need to figure it out themselves, and would google part of the comment "edge cases in floating number precision" and get a general idea of what is going on. I added an example to the comment, please let me know if you have any suggestions on how to make it more explanatory.

Reply:
yeah, I didn't mean an exhausting explanation but rather concrete examples, from which you as a reader can get a better idea where to start from and what class of edge-cases is relevant for this specific piece of logic, and tests are a suitable place for such edge-case examples, but no harm in adding such in the comment as well